server/src/internalClusterTest/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java

```diff
@@ -3346,77 +3342,6 @@ public void testSnapshotSucceedsAfterSnapshotFailure() throws Exception {
         assertEquals(SnapshotState.SUCCESS, getSnapshotsResponse.getSnapshots("test-repo-2").get(0).state());
     }
 
-    public void testSnapshotStatusOnFailedIndex() throws Exception {
```

Comment:
While this test used the old behavior to get a failed snapshot, it is still a useful test for listing good and bad snapshots, no?

Reply:
I guess the problem I had was that there was no way of creating a `FAILED` snapshot any longer and the whole premise of this test was to check that the status of a `FAILED` snapshot is returned properly from APIs.
Then again, as with the SLM test that I removed, let me see if I can create a BwC test for this by manipulating `RepositoryData` :)

Reply:
Alright, brought this back in a much simplified way to make sure we continue to be able to read the `FAILED` state. I think that's all we need here. Reading failed shard state we test in all kinds of places where we deal with `PARTIAL` snapshots so I think just faking a `0` shards, failed snapshot here is good enough.